Sources/Networking/Core/DownloadAPIManager.swift

```diff
@@ -0,0 +1,220 @@
+//
+//  DownloadAPIManager.swift
+//  
+//
+//  Created by Matej Molnár on 07.03.2023.
+//
+
+import Foundation
+import Combine
+
+/// Default Download API manager
+open class DownloadAPIManager: NSObject, Retryable {
+    private let requestAdapters: [RequestAdapting]
+    private let responseProcessors: [ResponseProcessing]
+    private let errorProcessors: [ErrorProcessing]
+    private var urlSession: URLSession!
+    private let sessionId: String
+    private var taskStateCancellables: [URLSessionTask: AnyCancellable] = [:]
+    private let downloadStateDictSubject = CurrentValueSubject<[URLSessionTask: URLSessionTask.DownloadState], Never>([:])
+    private var downloadStateDict = [URLSessionTask: URLSessionTask.DownloadState]() {
+        didSet {
+            downloadStateDictSubject.send(downloadStateDict)
+        }
+    }
+    
+    internal var retryCounter = Counter()
+    public var allTasks: [URLSessionDownloadTask] {
+        get async {
+            await urlSession.allTasks.compactMap { $0 as? URLSessionDownloadTask }
+        }
+    }
+    
+    public init(
+        urlSessionConfiguration: URLSessionConfiguration = .default,
+        requestAdapters: [RequestAdapting] = [],
+        responseProcessors: [ResponseProcessing] = [StatusCodeProcessor.shared],
+        errorProcessors: [ErrorProcessing] = []
+    ) {
+        /// generate session id in readable format
+        sessionId = Date().ISO8601Format()
+        
+        self.requestAdapters = requestAdapters
+        self.responseProcessors = responseProcessors
+        self.errorProcessors = errorProcessors
+        
+        super.init()
+        
+        urlSession = URLSession(
+            configuration: urlSessionConfiguration,
+            delegate: self,
+            delegateQueue: OperationQueue()
+        )
+    }
+}
+
+// MARK: Public API
+public extension DownloadAPIManager {
+    func downloadRequest(
+        _ endpoint: Requestable,
+        resumableData: Data? = nil,
+        retryConfiguration: RetryConfiguration?
+    ) async throws -> (URLSessionDownloadTask, Response) {
```

Comment:
Do you think may be worth to typealias return tuple into something easier to read? Similar we have Response = (Data, URLResponse)?